Adding Apps info to toast notification

nicosampler opened the pull request:
Ticket: #501 
depends on #601

gnosis-info commented:
Travis automatic deployment:
 https://pr621--safereact.review.gnosisdev.com/app

mikheevm commented on src/logic/notifications/notificationBuilder.js:
```diff
@@ -19,15 +21,26 @@ export type NotificationsQueue = {
-    moreConfirmationsNeeded: NOTIFICATIONS.TX_EXECUTED_MORE_CONFIRMATIONS_MSG,
-  },
-  afterExecutionError: NOTIFICATIONS.TX_FAILED_MSG,
+const customizeNotification = (notification: Notification, origin: string): Notification => {
```
nice customization options 😄  you can only add origin.

maybe a better name would be `setOriginForNotification` or `setNotificationOrigin`

nicosampler replied:
done

mikheevm commented on src/logic/notifications/notificationBuilder.js:
```diff
@@ -39,26 +52,30 @@ const waitingTransactionNotificationsQueue: NotificationsQueue = {
-    moreConfirmationsNeeded: NOTIFICATIONS.TX_CANCELLATION_EXECUTED_MSG,
-  },
-  afterExecutionError: NOTIFICATIONS.TX_FAILED_MSG,
+const getTancellationTxNotificationsQueue = (origin?: string): NotificationsQueue => {
```
typo `Tancellation`

nicosampler replied:
done

mikheevm commented on src/routes/safe/store/actions/createTransaction.js:
```diff
@@ -57,16 +57,16 @@ const createTransaction = ({
-  const nonce = await getNewTxNonce(txNonce, lastTx, safeInstance)
-  const isExecution = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+  const nonce = await getNewTxNonce(txNonce, lastTx, safeAddress)
+  const needsApproval = await doesTxNeedApproval(safeInstance, nonce, lastTx)
```
why was this changed from checking if tx can be executed right way to checking if tx needs approval?

nicosampler replied:
the base branch is outdated, this change should not be here.

nicosampler replied:
this change does not appear anymore

gnosis-info commented:
Travis automatic deployment:
 https://pr621--safereact.review.gnosisdev.com/app

gnosis-info commented:
Travis automatic deployment:
 https://pr621--safereact.review.gnosisdev.com/app

gnosis-info commented:
Travis automatic deployment:
 https://pr621--safereact.review.gnosisdev.com/app

francovenica commented:
Tested:
-Loaded a safe created in another pr with "Old compound tx" already performed. (Older meaning created before this PR)
-Older compound Tx are shown as "External App" with an icon similar to the one in the App Tab
-Accepted and rejected "External app" with no issues
-Checked that the notifications popping up show the "External App:" at the beginning of each one
-Created a new Tx in this PR, where the Compound investment is recognized as such.
-Approved and executed the Tx
-Rejected a Tx

1 - The icon in the list looks odd, not properly cropped.
Here is a side-by-side comparison with the icon in the tabs
![image.png](https://images.zenhubusercontent.com/5e14b8e3edee21d6f9aa6c8a/af93203a-c443-4d63-9d78-b9789397ff5d)
I assume the difference is that the one in the tabs is an icon while the one in the Tx list is an image.

2 -When created, the new Tx of compound are named "External App" by the notifications. 
![image.png](https://images.zenhubusercontent.com/5e14b8e3edee21d6f9aa6c8a/56917f4d-921c-4ad2-b4f0-71c2585962be)
This doesn't happen when you are approving, rejecting, executing the Tx, only the notifications that popUp when you are creating it: "Please sign the tx", "Tx pending", "Tx successfully created"...

3 - The notification "A pending transaction needs your confirmation" doesn't have the "External App:" or "Compound:" at the beginning like the others. This notification maybe is different (is a red notification instead of a white one) so let me know if this one is out of the scope of this ticket
![image.png](https://images.zenhubusercontent.com/5e14b8e3edee21d6f9aa6c8a/d39ec139-e8b8-428e-8fb8-7349797685f5)

nicosampler commented:
@francovenica 
1. Fixed
2. Fixed
3. Not possible to fix, the "pending TX requires confirmation" notification, it's a generic toast. I mean, you could have 5 pending Tx to sign and it will only show 1 notification. How it's implemented now, I don't have a way to recognize if the TX is from Apps or a standard one. Even, if you have more than 1 TX to sign, the message is in singular instead of plural.

gnosis-info commented:
Travis automatic deployment:
 https://pr621--safereact.review.gnosisdev.com/app

gnosis-info commented:
Travis automatic deployment:
 https://pr621--safereact.review.gnosisdev.com/app

francovenica commented:
1 and 2 fixed
3 - Will be left out then because the issue is not related to this PR.

Looks good to me. Approved.
